MPSGameLift/Code/Source/Components/UI/UiGameLiftConnectJsonMenuComponent.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/Component.h>
+#include <Multiplayer/IMultiplayer.h>
+
+#include <Request/AWSGameLiftJoinSessionRequest.h>
+#include <Multiplayer/Session/ISessionRequests.h>
+
+namespace MPSGameLift
+{
+    /*!
+     * \class UiGameLiftConnectJsonMenuComponent
```

Comment:
Is it relevant to expose Json in the name here? Seems like its leaking implementation details IMHO

Reply:
Hmm, how about "UiGameLiftConnectViaUserProvidedSessionIdMenuComponent"?

My goal is to differentiate it with other menu flows we plan to demo in the future like one where users just request to join any existing server, like how a typical game would work.

Reply:
Renamed to UiGameLiftConnectWithPlayerSessionData